src/error_estimation/adjoint_refinement_estimator.C

```diff
@@ -111,8 +112,25 @@ void AdjointRefinementEstimator::estimate_error (const System & _system,
 
   // Solve the adjoint problem(s) on the coarse FE space
   // Only if the user didn't already solve it for us
+  // If _adjoint_evaluation_physics pointer is not null, swap
+  // the current physics with the one held by _adjoint_evaluation_physics
+  // before assembling the adjoint problem
   if (!system.is_adjoint_already_solved())
-    system.adjoint_solve(_qoi_set);
+  {
+    // Swap the physics if needed, note that the current physics pointer
+    // held by the system will be pointed to by _adjoint_evalation_physics after swapping
+    const bool swapping_physics = _adjoint_evaluation_physics;
```

Comment:
Instead of recreating this bool a zillion times (with the same name but with two different meanings! we previously used it for _residual_evaluation_physics), can we either just directly test the physics pointers (they do implicitly convert to bool inside if()), or make two different bools, once each, with distinct names?

This annoys the compiler, too: with --enable-paranoid-warnings I get -Wshadow yelling at me.

Reply:
Makes sense.

Reply:
Was resolved in commit 3ee8a03494626d7e3c85561baccf59800a187ccb.